extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -16,12 +17,127 @@ import {
  */
 let ShoppingConfigDef;
 
+/**
+ * Validates string length of shopping config attributes
+ * @param {?string} field
+ * @param {?string} value
+ * @return {!Array<string>}
+ * @private
+ */
+function validateRequired(field, value) {
+  if (value == null) {
+    return [`${field} is required.`];
+  }
+}
+
+/**
+ * Validates string length of shopping config attributes
+ * @param {?string} field
+ * @param {?string} str
+ * @param {?number} maxLen
+ * @return {!Array<string>}
+ * @private
+ */
+function validateStringLength(field, str, maxLen = 100) {
+  if (str != null && str.length > maxLen) {
+    return [
+      'Length of ' +
+        field +
+        ' exceeds max length: ' +
+        str.length +
+        ' > ' +
+        maxLen,
+    ];
+  }
+}
+
+/**
+ * Validates number in shopping config attributes
+ * @param {?string} field
+ * @param {?number} number
+ * @private
+ * @return {!Array<string>}
+ */
+function validateNumber(field, number) {
+  if (number != null && isNaN(number)) {
+    return ['Value for field ' + field + ' is not a number'];
+  }
+}
+
+/**
+ * Validates url of shopping config attributes
+ * @param {?string} field
+ * @param {?string} url
+ * @private
+ * @return {!Array<string>}
+ */
+function validateURL(field, url) {
+  if (url == null) {
```

Comment:
Re: strict equality

Reply:
All that strict typing stuff is really important, you have a lot of typing approximations leading to various edge cases or incorrect method names in this file.